src/Microsoft.ML.Core/Utilities/IndentedTextWriterExtensions.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.CodeDom.Compiler;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Internal.Utilities
+{
+    public static class IndentedTextWriterExtensions
```

Comment:
Can this be internal and have a `[BestFriend]` attribute on it? I don't believe we want to ship this API publicly in our core assembly.